jobbergate-cli/jobbergate_cli/subapps/job_scripts/app.py

```diff
@@ -0,0 +1,294 @@
+"""
+
+import pathlib
+from typing import Any, Dict, List, Optional, cast
+
+import snick
+import typer
+
+from jobbergate_cli.constants import SortOrder
+from jobbergate_cli.exceptions import Abort, handle_abort
+from jobbergate_cli.render import StyleMapper, render_list_results, render_single_result, terminal_message
+from jobbergate_cli.requests import make_request
+from jobbergate_cli.schemas import (
+    JobbergateContext,
+    JobScriptCreateRequestData,
+    JobScriptResponse,
+    ListResponseEnvelope,
+)
+from jobbergate_cli.subapps.applications.tools import execute_application, fetch_application_data, load_application_data
+from jobbergate_cli.subapps.job_scripts.tools import fetch_job_script_data, validate_parameter_file
+from jobbergate_cli.subapps.job_submissions.app import HIDDEN_FIELDS as JOB_SUBMISSION_HIDDEN_FIELDS
+from jobbergate_cli.subapps.job_submissions.tools import create_job_submission
+
+
+# move hidden field logic to the API
+HIDDEN_FIELDS = [
+    "created_at",
+    "updated_at",
+    "job_script_data_as_string",
+]
+
+
+style_mapper = StyleMapper(
+    id="green",
+    job_script_name="cyan",
+)
+
+
+app = typer.Typer(help="Commands to interact with job scripts")
+
+
+@app.command()
+@handle_abort
+def list_all(
+    ctx: typer.Context,
+    show_all: bool = typer.Option(False, "--all", help="Show all job scripts, even the ones owned by others"),
+    search: Optional[str] = typer.Option(None, help="Apply a search term to results"),
+    sort_order: SortOrder = typer.Option(SortOrder.UNSORTED, help="Specify sort order"),
+    sort_field: Optional[str] = typer.Option(None, help="The field by which results should be sorted"),
+):
+    """
+    Show available job scripts
+    """
+    jg_ctx: JobbergateContext = ctx.obj
+
+    # Make static type checkers happy
+    assert jg_ctx is not None
+    assert jg_ctx.client is not None
+
+    params: Dict[str, Any] = dict(all=show_all)
+    if search is not None:
+        params["search"] = search
+    if sort_order is not SortOrder.UNSORTED:
+        params["sort_ascending"] = SortOrder is SortOrder.ASCENDING
+    if sort_field is not None:
+        params["sort_field"] = sort_field
+
+    envelope = cast(
+        ListResponseEnvelope,
+        make_request(
+            jg_ctx.client,
+            "/job-scripts",
+            "GET",
+            expected_status=200,
+            abort_message="Couldn't retrieve job scripts list from API",
+            support=True,
+            response_model=ListResponseEnvelope,
+            params=params,
+        ),
+    )
+    render_list_results(
+        jg_ctx,
+        envelope,
+        title="Job Scripts List",
+        style_mapper=style_mapper,
+        hidden_fields=HIDDEN_FIELDS,
+    )
+
+
+@app.command()
+@handle_abort
+def get_one(
+    ctx: typer.Context,
+    id: int = typer.Option(int, help="The specific id of the job script."),
+):
+    """
+    Get a single job script by id
```

Comment:
```suggestion
    Get a single job script by id.
```

Reply:
When I added it, it had 325 errors. I would like to add this, but in a future update.

I've created a task for it here: https://app.clickup.com/t/2md2f2u

Reply:
Original comment addressed in latest commit.